### PR TITLE
S1/wizard

### PR DESCRIPTION
**Pull request by graemehoffman**
Build out of the Wizard experience with initial copy. Add simple screen for choosing which field to add with explanation of what it is. Moving the related items fields to a settings checkbox. Unnecessary feidl cleanup. Diviner field Title is now the the label in the archive item edit screen

### CODE REVIEW

File: app/Admin/Settings.php [hunk 1]
```diff
@@ -59,6 +61,11 @@ public function browse_modal_field() {
 			->set_help_text( 'Modal displays by default mid size image, title, and copyright information' );
 	}
 
+	public function related_field() {
+		return Field::make( 'checkbox', self::FIELD_GENERAL_RELATED_FIELD, 'Activate Related Items Field on Archive Items' )
+			->set_help_text( 'Allows you to manually associate archive items to each other' );
```
Comment (lucatume) [inline app/Admin/Settings.php]: missing i18n here

File: app/Post_Types/Archive_Item/Post_Meta.php [hunk 2]
```diff
@@ -146,6 +146,19 @@ public function add_dynamic_fields(){
...
 			}
 		}
+
+		// add the related field
+		$related = carbon_get_theme_option(\Diviner\Admin\Settings::FIELD_GENERAL_RELATED_FIELD);
+		if ( $related ) {
+			$field = \Diviner\Post_Types\Diviner_Field\Types\Related_Field::render(
+				0,
+				'diviner_related',
+				'Related Archive Items',
```
Comment (lucatume) [inline app/Post_Types/Archive_Item/Post_Meta.php]: missing i18n here

File: app/Post_Types/Archive_Item/Post_Meta.php [hunk 3]
```diff
@@ -146,6 +146,19 @@ public function add_dynamic_fields(){
 
 			}
 		}
+
+		// add the related field
+		$related = carbon_get_theme_option(\Diviner\Admin\Settings::FIELD_GENERAL_RELATED_FIELD);
+		if ( $related ) {
+			$field = \Diviner\Post_Types\Diviner_Field\Types\Related_Field::render(
```
Comment (lucatume) [inline app/Post_Types/Archive_Item/Post_Meta.php]: all these fully-qualified references, `\Acme\Company\Package\SubPackage\EtCetera` could be simplified using `use`; this line could be rewritten like:

```php
use \Diviner\Post_Types\Diviner_Field\Types\Related_Field;

//...

RelatedField::render(...)
```

File: app/Post_Types/Archive_Item/Post_Meta.php [hunk 4]
```diff
@@ -146,6 +146,19 @@ public function add_dynamic_fields(){
...
+				0,
+				'diviner_related',
+				'Related Archive Items',
+				'Appears on each archive item single page as a slider'
```
Comment (lucatume) [inline app/Post_Types/Archive_Item/Post_Meta.php]: missing i18n here

File: app/Post_Types/Archive_Item/Post_Meta.php [hunk 5]
```diff
@@ -146,6 +146,19 @@ public function add_dynamic_fields(){
...
+			);
+			$dyn_fields[] = $field;
```
Comment (lucatume) [inline app/Post_Types/Archive_Item/Post_Meta.php]: let's be explicit here: `dynamic_fields` in place of `dyn_fields`, it's not that longer and more readable.

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 6]
```diff
@@ -5,13 +5,33 @@
...
 
+	const SLUG_WIZARD = 'diviner_wizard';
+
 	public function hooks() {
 		add_action( 'admin_menu', array( &$this,'rc_scd_register_menu') );
 		add_filter( 'admin_body_class', array( &$this,'admin_body_class') );
 		add_filter( 'gettext', array( &$this,'change_excerpt_text'), 10, 2 );
 
+		add_action( 'edit_form_after_title', array( &$this,'add_helper_text') );
+
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: too many blank lines here?

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 7]
```diff
@@ -5,13 +5,33 @@
...
+
+	}
+
+	function add_helper_text(){
+		global $post;
+		if( !empty($post) && $post->post_type !== Diviner_Field::NAME) {
+			return;
+		}
+		echo '<div>Label appearing on the Archive item Edit Screen</div>';
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: missing i18n here

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 8]
```diff
@@ -76,9 +106,18 @@ function rc_scd_register_menu() {
 
 		add_menu_page( 'Diviner Fields', 'Manage	 Diviner Fields', 'manage_options', 'diviner-manage-fields', array( &$this,'rc_scd_create_dashboard'), 'dashicons-admin-generic', 30 );
 
+		add_submenu_page(
+			null,           // -> Set to null - will hide menu link
+			'Page Title',    // -> Page Title
+			'Menu Title',   // -> Title that would otherwise appear in the menu
+			'administrator', // -> Capability level
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: this should be a capability, not a role name, e.g. `manage_options` or `install_plugins`

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 9]
```diff
@@ -76,9 +106,18 @@ function rc_scd_register_menu() {
 
 		add_menu_page( 'Diviner Fields', 'Manage	 Diviner Fields', 'manage_options', 'diviner-manage-fields', array( &$this,'rc_scd_create_dashboard'), 'dashicons-admin-generic', 30 );
 
+		add_submenu_page(
+			null,           // -> Set to null - will hide menu link
+			'Page Title',    // -> Page Title
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: missing i18n

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 10]
```diff
@@ -76,9 +106,18 @@ function rc_scd_register_menu() {
...
+			'Menu Title',   // -> Title that would otherwise appear in the menu
+			'administrator', // -> Capability level
+			self::SLUG_WIZARD,   // -> Still accessible via admin.php?page=menu_handle
+			array( &$this,'rc_scd_create_wizard') // -> To render the page
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: why the `&`? It's not needed. If we are targeting PHP 5.4 compatibility then let's use array compact syntax (`[]`)

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 11]
```diff
@@ -92,37 +131,144 @@ function rc_scd_create_dashboard() {
...
+				There are five kinds of fields:
+			</p>
+
+
+			<div class="field-select-wrap">
+				<h2>Text Field</h2>
+				<p>
+					<?php _e('Add a text field for a type of information you wish to assign to EACH archive item, but which will be completely different for each archive item. Example: serial number, catalog number, internal title, etc.' ); ?>
+				</p>
+				<a href="post-new.php?post_type=diviner_field&field_type=<?php echo CPT_Field::NAME; ?>" class="button button-primary button-hero">
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: ideally, in all these slugs `post_type=diviner_field` should be replaced by the use of a constant as field name is; this is not a big issue though.

File: app/Post_Types/Diviner_Field/AdminModifications.php [hunk 12]
```diff
@@ -92,37 +131,144 @@ function rc_scd_create_dashboard() {
...
 		</div>
 
-		<div class="wrap wrap-diviner wrap-diviner--preset" id="wrap-diviner--preset">
+
+
+
+		<?php
+	}
+
+	function rc_scd_create_dashboard() {
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/AdminModifications.php]: Is this still used? if so: why load all WordPress? My ignorance here.

File: app/Post_Types/Diviner_Field/PostMeta.php [hunk 13]
```diff
@@ -193,47 +189,68 @@ public function get_field_types() {
 			Select_Field::NAME => Select_Field::TITLE,
 			CPT_Field::NAME => CPT_Field::TITLE,
 		];
-		return Field::make( 'select', self::FIELD_TYPE, 'Type of field' )
+		$field =  Field::make( 'select', self::FIELD_TYPE, 'Type of field' )
+			->set_classes( self::FIELD_TYPE )
 			->add_options($types)
 			->set_help_text( 'What kind of field is this' );
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/PostMeta.php]: missing i18n here and below

File: app/Post_Types/Diviner_Field/Preset_Fields_List_Table.php [hunk 14]
```diff
@@ -12,25 +12,30 @@
...
+
+    public function is_empty() {
+        return (bool) ( isset( $fields ) && count( $fields ) > 0 );
+    }
 
 	public function get_fields()
 	{
 		$args = [
-			'post_type' => \Diviner\Post_Types\Diviner_Field\Diviner_Field::NAME
+			'post_type' => \Diviner\Post_Types\Diviner_Field\Diviner_Field::NAME,
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/Preset_Fields_List_Table.php]: Use `use` and shorten all these fully qualified class calls

File: docs-src/content/getting-started.md [hunk 15]
```diff
@@ -3,7 +3,7 @@ title: "Getting Started"
 draft: false
 ---
 
-**How the three pines shake!** Pines are the hardest sort of tree to live when shifted to any other soil, and here there's none but the crew's cursed clay. _Steady, helmsman! steady._ This is the sort of weather when brave hearts snap ashore, and keeled hulls split at sea. Our captain has his birthmark; look yonder, boys, [there's another in the sky&mdash;lurid-like](#), ye see, all else pitch black.
+**How the three pines shake asdhkhasdhkas!** Pines are the hardest sort of tree to live when shifted to any other soil, and here there's none but the crew's cursed clay. _Steady, helmsman! steady._ This is the sort of weather when brave hearts snap ashore, and keeled hulls split at sea. Our captain has his birthmark; look yonder, boys, [there's another in the sky&mdash;lurid-like](#), ye see, all else pitch black.
```
Comment (lucatume) [inline docs-src/content/getting-started.md]: ???

File: app/Admin/Settings.php (hunk 1 above)
Comment (graemehoffman) [reply app/Admin/Settings.php]: Done... I am lazy

File: app/Post_Types/Archive_Item/Post_Meta.php (hunk 2 above)
Comment (graemehoffman) [reply app/Post_Types/Archive_Item/Post_Meta.php]: Done

File: app/Post_Types/Archive_Item/Post_Meta.php (hunk 3 above)
Comment (graemehoffman) [reply app/Post_Types/Archive_Item/Post_Meta.php]: Done

File: app/Post_Types/Archive_Item/Post_Meta.php (hunk 4 above)
Comment (graemehoffman) [reply app/Post_Types/Archive_Item/Post_Meta.php]: Done

File: app/Post_Types/Archive_Item/Post_Meta.php (hunk 5 above)
Comment (graemehoffman) [reply app/Post_Types/Archive_Item/Post_Meta.php]: Done

File: app/Post_Types/Diviner_Field/AdminModifications.php (hunk 11 above)
Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/AdminModifications.php]: Done

File: app/Post_Types/Diviner_Field/AdminModifications.php (hunk 12 above)
Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/AdminModifications.php]: Yes, the function is used but I removed those includes.

File: app/Post_Types/Diviner_Field/PostMeta.php (hunk 13 above)
Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/PostMeta.php]: Done

File: app/Post_Types/Diviner_Field/Preset_Fields_List_Table.php (hunk 14 above)
Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/Preset_Fields_List_Table.php]: done

File: docs-src/content/getting-started.md (hunk 15 above)
Comment (graemehoffman) [reply docs-src/content/getting-started.md]: This is for placement only for the git pages documentation. Uses Hugo and markdown in flat files.  We may end up removing it. For now you can just ignore

**Pull request merged**